bin/portal-bridge/src/bridge/e2hs.rs

```diff
@@ -0,0 +1,439 @@
+    ) -> anyhow::Result<()> {
+        let header_number = header_with_proof.header.number;
+        let hwp_by_number_content_key =
+            HistoryContentKey::new_block_header_by_number(header_number);
+        let hwp_content_value = HistoryContentValue::BlockHeaderWithProof(header_with_proof);
+        if let Err(err) = gossip_history_content(
+            portal_client,
+            hwp_by_number_content_key.clone(),
+            hwp_content_value,
+            block_stats,
+        )
+        .await
+        {
+            error!(
+                "Failed to gossip history content key: {:?} - {:?}",
+                hwp_by_number_content_key, err
+            );
+        }
+        Ok(())
+    }
+
+    async fn gossip_body(
+        body: BlockBody,
+        header_hash: B256,
+        portal_client: HttpClient,
+        block_stats: Arc<Mutex<HistoryBlockStats>>,
+    ) -> anyhow::Result<()> {
+        let body_content_key = HistoryContentKey::new_block_body(header_hash);
+        let body_content_value = HistoryContentValue::BlockBody(body);
+        if let Err(err) = gossip_history_content(
+            portal_client,
+            body_content_key.clone(),
+            body_content_value,
+            block_stats,
+        )
+        .await
+        {
+            error!(
+                "Failed to gossip history content key: {:?} - {:?}",
+                body_content_key, err
+            );
+        }
+        Ok(())
+    }
+
+    async fn gossip_receipts(
+        receipts: Receipts,
+        header_hash: B256,
+        portal_client: HttpClient,
+        block_stats: Arc<Mutex<HistoryBlockStats>>,
+    ) -> anyhow::Result<()> {
+        let receipts_content_key = HistoryContentKey::new_block_receipts(header_hash);
+        let receipts_content_value = HistoryContentValue::Receipts(receipts);
+        if let Err(err) = gossip_history_content(
+            portal_client,
+            receipts_content_key.clone(),
+            receipts_content_value,
+            block_stats,
+        )
+        .await
+        {
+            error!(
+                "Failed to gossip history content key: {:?} - {:?}",
+                receipts_content_key, err
+            );
+        }
+        Ok(())
+    }
+
+    fn validate_block_tuple(&self, block_tuple: &BlockTuple) -> anyhow::Result<()> {
+        self.header_validator
+            .validate_header_with_proof(&block_tuple.header_with_proof.header_with_proof)?;
+        let receipts = &block_tuple.receipts.receipts;
+        let receipts_root = receipts.root();
+        ensure!(
+            receipts_root
+                == block_tuple
+                    .header_with_proof
+                    .header_with_proof
+                    .header
+                    .receipts_root,
+            "Receipts root mismatch"
+        );
+        let body = &block_tuple.body.body;
+        body.validate_against_header(&block_tuple.header_with_proof.header_with_proof.header)?;
+        Ok(())
+    }
```

Comment:
```suggestion
    fn validate_block_tuple(&self, block_tuple: &BlockTuple) -> anyhow::Result<()> {
        let header_with_proof = &block_tuple.header_with_proof.header_with_proof;
        self.header_validator
            .validate_header_with_proof(header_with_proof)?;
            
        let body = &block_tuple.body.body;
        body.validate_against_header(&header_with_proof.header)?;            
        
        let receipts = &block_tuple.receipts.receipts;
        ensure!(
            receipts.root()
                == header_with_proof
                    .header
                    .receipts_root,
            "Receipts root mismatch"
        );        
        Ok(())
    }
```

I think ordering it like this would be a bit more readable. Also I don't think we need the extra variable `receipts_root`

Idk up to you